app/Http/Controllers/DealController.php

```
@@ -2194,11 +2207,25 @@ protected function changeBulkStatus($request, ?array $editableIds = null)
+        // and notify their agents/watchers explicitly after the update (in-app
+        // only — mail stays suppressed by the bulk-action container flag).
+        $deals = Deal::whereIn('id', $editableIds)->get();
+
         if ($stage->slug === 'win' || $stage->slug === 'lost') {
             Deal::whereIn('id', $editableIds)->whereNull('close_date')->update(['close_date' => now()->format('Y-m-d')]);
         }
 
         Deal::whereIn('id', $editableIds)->update(['pipeline_stage_id' => $newStatus]);
+
+        $notificationService = app(\App\Services\DealNotificationService::class);
+        foreach ($deals as $deal) {
```

PHPStan (2.2.7)

[error] Argument of an invalid type Illuminate\Support\Collection supplied for foreach, only iterables are supported.


(foreach.nonIterable)